languages/definitions/Arabic.yml

```diff
@@ -0,0 +1,16 @@
+---
+locale: ar-JO
+dictionaryFile: ar-utf8.csv
+abcString: ابت
+hasUpperCase: no
+layout:
+  - [SPECIAL] # 0
+  - [PUNCTUATION,؟,ى,ئ] # 1
```

Comment:
Letters on 1-key are not supported and may cause strange side effects while typing. It may be possible to make it work, since English, French, Hebrew and Ukrainian sort of use apostrophes and dashes as letters, meaning there are words with 1-key in their dictionaries, but there is no 100% guarantee that it will work properly.

Also, I don't see them included in standard hardware layouts, for example, [this Nokia](https://www.alamy.com/arabic-keyboard-on-cell-phone-in-hand-image343639432.html).

I suppose, they don't have a standard position? Would it be possible to move them to another key?

Reply:
i actually have an old samsung candy bar dumbphone from the mid 2000s in arabic and i have referenced it for the letters on the 3 key, the physical keyboard is identical to the picture you have linked but in actuality in the software the 3 key has a massive amount of letters linked to it, mostly different forms of alif (the first letter in the alphabet). the arabic keyboard is mapped from right to left, so physically the order of the alphabet on the keyboard maps to, in order, 3 2 6 5 4 9 8 7, the yml file has been fully mapped correctly to the standard layout with the exception being the 1 key as that needs to be fixed once i know where to add the punctuation list. i have the phone so please ask any additional questions about how the original arabic keyboard works on these old arab T9 phones. the reason i mentioned justification is that the TT9 software in abc mode shows the letter selection left to right in english when keys are being selected and was just wondering if it would automatically switch to right to left selections in arabic mode.

Reply:
also I dont see the options for this in any of the files but even though the physical arabic keypad shows 1234567890, it types out as ١٢٣٤٥٦٧٨٩٠ (in that order, in arabic the numbers read from left to right just like english, even though the rest of the language is read right to left)